routes/beetles.py

```diff
@@ -0,0 +1,578 @@
+            di["1988-2017"]["Daymet"]["Historical"][snowpack][
+                "percent-medium-risk"
+            ] = beetle_percents[0][0][0][sni][2]
+            di["1988-2017"]["Daymet"]["Historical"][snowpack][
+                "percent-high-risk"
+            ] = beetle_percents[0][0][0][sni][3]
+
+    # Gather predicted risk levels for future eras
+    # We use
+    for ei, mod_li in enumerate(beetle_resp[1:]):
+        era = dim_encodings["era"][ei]
+        di[era] = dict()
+        for mi, sc_li in enumerate(mod_li[1:]):
+            model = dim_encodings["model"][mi]
+            di[era][model] = dict()
+            for si, sn_li in enumerate(sc_li[1:]):
+                scenario = dim_encodings["scenario"][si]
+                di[era][model][scenario] = dict()
+                for sni, risk_level in enumerate(sn_li):
+                    snowpack = dim_encodings["snowpack"][sni]
+                    di[era][model][scenario][snowpack] = dict()
+                    di[era][model][scenario][snowpack]["beetle-risk"] = dim_encodings[
+                        "beetle_risk"
+                    ][int(risk_level)]
+                    if beetle_percents is not None:
+                        di[era][model][scenario][snowpack][
+                            "percent-low-risk"
+                        ] = beetle_percents[ei + 1][mi + 1][si + 1][sni][1]
+                        di[era][model][scenario][snowpack][
+                            "percent-medium-risk"
+                        ] = beetle_percents[ei + 1][mi + 1][si + 1][sni][2]
+                        di[era][model][scenario][snowpack][
+                            "percent-high-risk"
+                        ] = beetle_percents[ei + 1][mi + 1][si + 1][sni][3]
+
+    return di
+
+
+def summarize_within_poly_marr(ds, poly_mask_arr, bandname="Gray"):
+    """Summarize a single Data Variable of a xarray.DataSet within a polygon.
+    Return the results as a nested dict.
+
+    Args:
+        ds (xarray.DataSet): DataSet with "Gray" as variable of
+            interest
+        poly_mask_arr (numpy.ma.core.MaskedArra): a masked array masking the cells intersecting
+            the polygon of interest
+        dim_encodings (dict): nested dictionary of thematic key value pairs that characterize the
+            data and map integer data coordinates to models, scenarios, eras, snowpacks, etc.
+        bandname (str): name of variable in ds, defaults to "Gray" for Rasdaman coverages where
+            the name is not given at ingest
+
+    Returns:
+        Nested dict of results for all non-X/Y axis combinations,
+    """
+
+    # operates on underlying DataArray
+    da = ds[bandname]
+
+    # get axis (dimension) names and make list of all coordinate combinations
+    all_dims = da.dims
+    dimnames = [dimname for dimname in all_dims if dimname not in ("X", "Y")]
+    iter_coords = list(
+        itertools.product(*[list(ds[dimname].values) for dimname in dimnames])
+    )
```

Comment:
I'm going to create a ticket for this code block because this is a piece that gets duplicated across many endpoints! Nothing to change right now though for this endpoint / PR.